base/strings/string.jl

```diff
@@ -51,7 +51,7 @@ convert(::Type{String}, v::Vector{UInt8}) = String(v)
 pointer(s::String) = unsafe_convert(Ptr{UInt8}, s)
 pointer(s::String, i::Integer) = pointer(s)+(i-1)
 
-sizeof(s::String) = s.len
+sizeof(s::String) = unsafe_load(convert(Ptr{Int},data_pointer_from_objref(s)))
```

Comment:
with https://github.com/JuliaLang/julia/pull/22425 this can be `Core.sizeof(s)`.